source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,253 @@
+see :ref:`csharp-retrieve-additional-information`.
+
+Find All Documents
+~~~~~~~~~~~~~~~~~~
+
+To find all documents in a collection, pass an empty
+``BsonDocument`` as a query filter. Then chain the ``ToList()`` or ``ToListAsync()``
+method to return the documents as a ``List`` object, or the ``ToCursor()`` or ``ToCursorAsync()``
+method to return the documents as a cursor.
+
+To see a fully runnable example of using the ``Find()`` method to find all documents, see 
+:ref:`csharp-retrieve-additional-information`.
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``Find()`` method by passing
+a ``FindOptions`` object.
+
+You can configure the commonly used options with the following methods:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``BatchSize`` 
+     - | Gets or sets the number of documents to return per batch.
+
+   * - ``Collation`` 
+     - | Sets the collation options. 
+
+   * - ``Comment`` 
+     - | Sets the comment to the query.
+
+   * - ``Hint`` 
+     - | Sets the hint for which index to use.
+
+   * - ``MaxTime`` 
+     - | Sets the maximum execution time on the server for this operation.
+
+To see a full list of available options, see 
+`FindOptions Properties <{+api-root+}/Properties_T_MongoDB_Driver_FindOptions.htm>`__.
+
+Example
+~~~~~~~
+
+The following example configures the behavior of the ``Find()`` method with a 
+``FindOptions`` object. The example performs the following actions:
+
+- Finds all documents with "Pizza" in the ``cuisine`` field
+- Sets the ``BatchSize`` to ``3``
+- Stores the results in a cursor
+- Prints the number of documents currently held in the cursor
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var filter = Builders<Restaurant>.Filter.Eq("cuisine", "Pizza");
+
+      var findOptions = new FindOptions
+      {
+         BatchSize = 3
+      };
+
+      using (var cursor = _restaurantsCollection.Find(filter, findOptions).ToCursor())
+      {
+         cursor.MoveNext();
+         WriteLine($"Number of documents in cursor: {cursor.Current.Count()}");
+      }
+
+   .. output::
+      
+      Number of documents in cursor: 3
+
+.. tip:: Clean Up
+
+   Create a cursor with a `using statement <https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement>`__ to
+   automatically invoke the 
+   `Dispose() <https://learn.microsoft.com/en-us/dotnet/api/system.idisposable.dispose?view=net-7.0>`__
+   method once the cursor is no longer in use.
+
+
+.. _csharp-retrieve-additional-information:
+
+Additional Information
+----------------------
+
+.. TODO: Links to creating filter with Builders and LINQ
+
+To learn more about query filters, see :ref:`csharp-specify-query`
+
+For fully runnable examples of the ``Find()`` method, see the 
+:ref:`Usage Examples <csharp-find-one>`.
```

Comment:
Suggestion: Could the destination of this link be made more precise?
Issue: For -> "To view" to add imperative mood

```suggestion
To view a fully runnable example of the ``Find()`` method, see the 
:ref:`<csharp-find-one>` page.
```